GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/utils/Utils.java

```diff
@@ -95,4 +92,35 @@ private void onRequestPermissionsResult(int requestCode, String[] permissions,
     public static boolean isGpsEnabled(LocationManager locationManager) {
         return locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
     }
+
+    /**
+     * Start capture via intent
+     */
+    public static void startCaptureViaIntent(GpsDataCaptureService gpsDataCaptureService,
+            Intent intent) {
+        String action = intent.getAction();
+        Log.d(TAG, "Intent action: " + action);
+
+        if(gpsDataCaptureService != null) {
+            //Extra the location api type
+            LocationApiType locationApiType = LocationApiType.LOCATIONMANAGER;
+            if (intent.hasExtra("fused_location_type")) {
+                boolean type_from_intent = intent.getBooleanExtra("fused_location_type", false);
+                Log.d(TAG, "fused_location_type: " + type_from_intent);
+
+                if (type_from_intent) {
```

Comment:
type_from_intent : snake  names are not used in Java. Also if boolean it should be in predicate form, isA or hasA. 
will be more readable if type_from_intent is not boolean but enum:
if (typeFromIntent == FUSED_API) ...

but if working with boolean name should suggest what property we are checking
if (isTypeFromIntentFused) { ...